src/libraries/System.ComponentModel.TypeConverter/tests/Design/DesigntimeLicenseContextSerializerTests.cs

```diff
@@ -0,0 +1,146 @@
+            AppContext.SetSwitch(enableBinaryFormatterInTypeConverter, true);
+            var context = new DesigntimeLicenseContext();
+            context.SetSavedLicenseKey(typeof(int), key);
+            var assembly = typeof(DesigntimeLicenseContextSerializer).Assembly;
+            using (MemoryStream stream = new MemoryStream())
+            {
+                long position = stream.Position;
+                DesigntimeLicenseContextSerializer.Serialize(stream, key, context);
+                stream.Seek(position, SeekOrigin.Begin);
+                VerifyStreamFormatting(stream);
+
+                string tempPath = Path.GetTempPath();
+                using (FileStream outStream = File.Create(Path.Combine(tempPath, "_temp_SerializeWithBinaryFormatter_DeserializeWithBinaryWriter")))
+                {
+                    stream.Seek(position, SeekOrigin.Begin);
+                    stream.CopyTo(outStream);
+                }
+            }
+
+            RemoteExecutor.Invoke((key) =>
+            {
+                var assembly = typeof(DesigntimeLicenseContextSerializer).Assembly;
+                Type runtimeLicenseContextType = assembly.GetType("System.ComponentModel.Design.RuntimeLicenseContext");
+                Assert.NotNull(runtimeLicenseContextType);
+                object runtimeLicenseContext = Activator.CreateInstance(runtimeLicenseContextType);
+                Assert.NotNull(runtimeLicenseContext);
+                FieldInfo _savedLicenseKeys = runtimeLicenseContextType.GetField("_savedLicenseKeys", BindingFlags.NonPublic | BindingFlags.Instance);
+                Assert.NotNull(_savedLicenseKeys);
+                _savedLicenseKeys.SetValue(runtimeLicenseContext, new Hashtable());
+
+                Type designtimeLicenseContextSerializer = assembly.GetType("System.ComponentModel.Design.DesigntimeLicenseContextSerializer");
+                Assert.NotNull(designtimeLicenseContextSerializer);
+                MethodInfo deserializeMethod = designtimeLicenseContextSerializer.GetMethod("Deserialize", BindingFlags.NonPublic | BindingFlags.Static);
+                Assert.NotNull(deserializeMethod);
+
+                string tempPath = Path.GetTempPath();
+                using (FileStream stream = File.Open(Path.Combine(tempPath, "_temp_SerializeWithBinaryFormatter_DeserializeWithBinaryWriter"), FileMode.Open))
+                {
+                    try
+                    {
+                        deserializeMethod.Invoke(null, new object[] { stream, key, runtimeLicenseContext });
+                    }
+                    catch (TargetInvocationException exception)
+                    {
+                        Exception baseException = exception.GetBaseException();
+                        Assert.IsType<NotSupportedException>(baseException);
+                    }
+                }
+                File.Delete(Path.Combine(tempPath, "_temp_SerializeWithBinaryFormatter_DeserializeWithBinaryWriter"));
```

Comment:
Should do this in a `try-finally` outside of the RemoteExecutor - at the same level as what created the file.

Reply:
Why `try-finally`? If there is an exception because the file is not found for some reason, I want that to bubble up.